syntax = "proto3";

package orbiterkep;

message ParamBounds {

  double lb = 1;
  double ub = 2;

}

message ParamPaGMO {

  int32 n_isl = 1;
  int32 population = 2;

  int32 n_gen = 3;
  int32 mf = 4;
  double mr = 5;
}

message Parameters {

  repeated string planets = 1;
  repeated string single_objective_algos = 2;
  repeated string multi_objective_algos = 3;

  ParamBounds t0 = 4;
  ParamBounds tof = 5;
  ParamBounds vinf = 6;

  string problem = 7;
  int32 n_trials = 8;

  double dep_altitude = 10;
  double arr_altitude = 11;

  double max_deltav = 12;

  bool add_dep_vinf = 13;
  bool add_arr_vinf = 14;

  bool use_spice = 15;
  bool circularize = 16;

  bool multi_objective = 17;

  bool use_db = 18;

  ParamPaGMO pagmo = 19;
}
